Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -197,6 +199,34 @@ template <class Type> class TableColumn : public API::Column {
   /// Re-arrange values in this column according to indices in indexVec
   void sortValues(const std::vector<size_t> &indexVec) override;
 
+  bool equals(Column *otherColumn, double tolerance) const override {
+    auto convertedOtherColumn = dynamic_cast<TableColumn *>(otherColumn);
+    if (!convertedOtherColumn) {
+      return false;
+    }
+    auto type = this->type();
+    if (convertedOtherColumn->size() != this->size() ||
+        convertedOtherColumn->type() != type) {
```

Comment:
Comparing the `hash` above would, I think, make `convertedOtherColumn->type() != type` unnecessary?